game/test/ecs/tools/interaction/InteractionToolTest.java

```diff
@@ -0,0 +1,232 @@
+import level.tools.DesignLabel;
+import level.tools.LevelElement;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import starter.Game;
+import testinghelper.SimpleCounter;
+import tools.Point;
+
+public class InteractionToolTest {
+
+    private static ILevel prepareLevel() {
+        LevelElement[][] layout = new LevelElement[5][5];
+        for (int y = 0; y < 5; y++) {
+            for (int x = 0; x < 5; x++) {
+                layout[y][x] = LevelElement.FLOOR;
+            }
+        }
+        return new TileLevel(layout, DesignLabel.DEFAULT);
+    }
+
+    /**
+     * Helper Method which prepares a mock, of the Hero which can´t be created using the
+     * constructor.
+     *
+     * @param havingpc if the Hero should have the PositionComponent
+     * @return the Mocked Hero
+     */
+    private static Hero fullMockedHero(boolean havingpc) {
+        Hero mock = Mockito.mock(Hero.class);
+        Optional<Component> pc;
+        if (havingpc) {
+            pc = Optional.of(new PositionComponent(mock, new Point(0, 0)));
+        } else {
+            pc = Optional.empty();
+        }
+        Mockito.when(mock.getComponent(PositionComponent.class)).thenReturn(pc);
+
+        return mock;
+    }
+
+    /** cleanup to reset static Attributes from Game used by the InteractionTool */
+    private static void cleanup() {
+        Game.entities.clear();
+        Game.hero = null;
+        Game.currentLevel = null;
+    }
+
+    /** Tests the functionality when the Hero does not have the PositionComponent */
+    @Test
+    public void interactWithClosestInteractableHeroMissingPositionComponent() {
+        cleanup();
+        Game.hero = fullMockedHero(false);
+        Game.currentLevel = prepareLevel();
+
+        MissingComponentException e =
+                Assert.assertThrows(
+                        MissingComponentException.class,
+                        () -> InteractionTool.interactWithClosestInteractable(Game.hero));
+        Assert.assertTrue(e.getMessage().contains(InteractionTool.class.getName()));
+        Assert.assertTrue(e.getMessage().contains(Hero.class.getName()));
+        Assert.assertTrue(e.getMessage().contains(PositionComponent.class.getName()));
+        cleanup();
+    }
+
+    /** Tests the functionality when there are no Entities in the Game */
+    @Test
+    public void interactWithClosestInteractableNoEntities() {
+        cleanup();
+        Game.hero = fullMockedHero(true);
+        Game.currentLevel = prepareLevel();
+        InteractionTool.interactWithClosestInteractable(Game.hero);
+        cleanup();
+    }
+
+    /**
+     * Tests the functionality when there are no Entities with the interactionComponent in the Game
+     */
+    @Test
+    public void interactWithClosestInteractableNoInteractable() {
+        cleanup();
+        Game.hero = fullMockedHero(true);
+        Game.currentLevel = prepareLevel();
+        Game.entities.add(Game.hero);
```

Comment:
`Game.entities` ist ein `Set`. Daher kann der Hero nur einmal hinzugefügt werden. Oder führt Mockito den Konstruktor von `Entity` nicht aus, sodass `Game.hero` noch nicht zu dem Set hinzugefügt wurde?
Im Konstruktor von `Entity` wird die erzeugte Entität direkt zum Set hinzugefügt.

Reply:
Ich füge jetzt mal nicht überall wo das vorkommt ein Kommentar hinzu, sind schon genug hier in dem review.

Reply:
Also Mocks sind nur pseudo Kopien einer Klasse. Sie sind nicht die Klasse daher wird nicht im ctor Game.entities.add(this) aufgerufen. Daher muss es hinzugefügt werden. Bei jedem anderen Test habe ich das hinzufügen entfernt, aber hier lasse ich ihn drinnen da es sich hier einfach um die gegebene Entität handelt.